src/analyses/natural_loops.h

```diff
@@ -241,7 +241,12 @@ void natural_loops_templatet<P, T>::compute_natural_loop(T m, T n)
   std::stack<T> stack;
 
   auto insert_result = loop_map.emplace(n, natural_loopt{*this});
-  INVARIANT(insert_result.second, "each loop head should only be visited once");
+  // Note the emplace *may* access a loop that already exists: this happens when
+  // a given header has more than one incoming edge, such as
+  // head: if(x) goto head; else goto head;
+  // In this case this compute routine is run twice, one for each backedge, with
+  // each adding whatever instructions can reach this 'm' (the program point
+  // that branches to the loop header, 'n').
```

Comment:
I suspect this also happens for loops with `continue` statements? That might be a more natural example than one involving gotos.

Reply:
Deliberately didn't do that because `goto_convert` is liable to turn `continue;` into a branch to the bottom (leading directly into the backedge), similarly `javac`